File: oracle-next-education/front-end-f2t2-one/JavaScript-explorando-a-linguagem/Conta/ContaCorrente.js
import { Conta } from "./Conta.js";

export class ContaCorrente extends Conta {
    static numeroDeContas = 0;
    constructor(cliente, agencia){
        super(0, cliente, agencia);
        ContaCorrente.numeroDeContas++;
    }

    //sobreescrevendo o metodo sacar da classe herdada;
    sacar(valor){
        const taxa = 1.1;
        return super._sacar(valor, taxa);
    }
}

File: alura/ano-2023/java/bytebankHerdadoConta/src/br/com/bytebank/banco/model/ContaCorrente.java
package br.com.bytebank.banco.model;

import br.com.bytebank.banco.exception.SacaException;

public class ContaCorrente extends Conta {
	public ContaCorrente(int agencia, int numero) {
		super(agencia, numero);
	}

	public void sacar(double valor) throws SacaException {
		double taxaTransferencia = 0.2;
		super.sacar(valor + taxaTransferencia);
	}

	@Override
	public void depositar(double valor) {
		super.saldo += valor;
	}

	@Override
	public String toString() {
		return "Conta Corrente => " + super.toString();
	}

}
